app/jsx/components/Search2Comp.jsx

```diff
@@ -3,72 +3,89 @@
 import React from 'react'
 import Form from 'react-router-form'
 
-class CustomRadioInput extends React.Component {
-  render() {
-    return (
-      <input type="radio" id="c-search2__refine-campus" name={this.props.name} value={this.props.value} onClick={this.props.handleRadioSelect}/>
-  )}
-}
+// var customRadioLabel = ""
+// if (["oru", "series", "monograph_series", "seminar_series", "special"].includes(p.type)) {
+//    customRadioLabel = "This department"
+// }
+// if (p.type == 'journal') {
+//    customRadioLabel = "This Journal"
+// }
+// if (p.type == 'campus') {
+//    customRadioLabel = "This campus"
+// }
 
 class SearchControls extends React.Component {
-  labelCustomRadio(name) {
-    return <label htmlFor="c-search_2_refine-campus">{name}</label>
-  }
-
   render() {
     let p = this.props,
-        customRadioName = "",
-        customRadioLabel = ""
-    if (["oru", "series", "monograph_series", "seminar_series", "special"].includes(p.type)) {
-       customRadioName = "departments"
-       customRadioLabel = "This department"
-    }
-    if (p.type == 'journal') {
-       customRadioName = "journals"
-       customRadioLabel = "This Journal"
-    }
-    if (p.type == 'campus') {
-       customRadioName = "campuses"
-       customRadioLabel = "This campus"
+      searchUnit = null;
+
+    if (p.unitID) {
+      searchUnit = (
+        <span>
+          <input type="radio" id="c-search2__refine-campus" name="searchType" value={p.unitID} onClick={this.props.handleRadioSelect} />
+          <label htmlFor="c-search_2_refine-campus">This {p.label}</label>
+        </span>
+      );
     }
+
     return (
     <div className={this.props.refineActive ? "c-search2__refine--active" : "c-search2__refine"}>
-      <input type="radio" id="c-search2__refine-eschol" name={customRadioName} value="" onClick={()=>this.props.handleRadioSelect()} defaultChecked={true}/>
+      <input type="radio" id="c-search2__refine-eschol" name="searchType" value="eScholarship" onClick={this.props.handleRadioSelect} defaultChecked={true}/>
       <label htmlFor="c-search2__refine-eschol">All of eScholarship</label>
-      {p.unitID ? <CustomRadioInput name={customRadioName}
-                                   value={p.unitID}
-                                   handleRadioSelect={()=>this.props.handleRadioSelect()} /> : null }
-      {p.unitID ? this.labelCustomRadio(customRadioLabel) : null}
+      { searchUnit }
     </div>
   )}
 }
 
 class SearchComp2 extends React.Component {
   constructor(props){
     super(props)
-    this.state = {refineActive: false}
+
+    if (["series", "monograph_series", "seminar_series", "special"].includes(props.type)) {
+      this.label = "Series"
+    } else if (props.type == "oru") {
+      this.label = "Department"
+    } else if (props.type == "journal") {
+      this.label = "Journal"
+    } else if (props.type == "campus") {
+      this.label = "Campus"
+    }
```

Comment:
It feels like the computation of this.label should be done inside render(), or at least in componentWillReceiveProps(). Because, what if the props change when the user switches to a new unit -- the label might get out of date.